airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -716,6 +636,80 @@ class Issues(IncrementalGithubStream):
     }
 
 
+class PullRequestSubStream(SemiIncrementalGithubStream, ABC):
+    def __init__(self, parent: PullRequests, **kwargs):
+        self._parent_stream = parent
+        super().__init__(**kwargs)
+
+    use_cache = False
+
+    @property
+    def state_checkpoint_interval(self) -> Optional[int]:
+        return None
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        for stream_slice in super().stream_slices(**kwargs):
+            pull_requests = list(self._parent_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice, stream_state=stream_state))
+            if self._parent_stream.is_sorted_descending:
+                pull_requests.reverse()
+            for pull_request in pull_requests:
+                yield {"pull_request_number": pull_request["number"], "repository": stream_slice["repository"], self.cursor_field: pull_request[self.cursor_field]}
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        yield from super(SemiIncrementalGithubStream, self).read_records(
+            sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+        )
```

Comment:
Why do we need this function?

Reply:
To skip the start_point_map and cursor_field logic in `SemiIncrementalGithubStream.read_records()`, since the parent `PullRequests` stream will have already handled that when fetching the pull request records.